### ISSUE TITLE
[Backend Rewrite] Investigate possibilities for rewriting collaborative editing

### ISSUE THREAD

**Issue by RobbinBaauw** [finitum/CSHub#217]
This issue cannot yet be subdivided as we need to figure out how to move on. A few possibilities
- Switch to the TipTap editor (https://github.com/scrumpy/tiptap), see #190. This uses ProseMirror, which provides collaborative editing (https://prosemirror.net/docs/guide/#collab). I think this is the best option provided we figure out how to write the backend in Rust.
- Create an editor from scratch using Rust & WASM, we could use https://github.com/tcr/edit-text as a baseline. Note that this would require quite some client work.
- Look at Rust implementations of CRDTs and figure out how we can use those with a frontend editor
- Completely discard the Rust rewrite and rewrite the backend for this in another language, for example Go or TypeScript

Let's discuss in the comments :)

**Comment by NULLx76** [comment]
> Completely discard the Rust rewrite and rewrite the backend for this in another language, for example Go or TypeScript.

The backend language _shouldn't_ matter that much besides some library availability I would imagine.

**Comment by RobbinBaauw** [comment]
Yes, if there are amazing libraries on other platforms we should consider those as well imo
